src/linter.ts

```diff
@@ -0,0 +1,69 @@
+import * as core from '@actions/core'
+import { readFile } from 'fs/promises'
+import type { LinterResponse } from './types'
+import fetch from 'node-fetch'
+
+export async function lintFiles(
+  files: string[],
+  apiKey: string,
+  axeLinterUrl: string,
+  linterConfig: Record<string, unknown>
+): Promise<number> {
+  let totalErrors = 0
+
+  for (const file of files) {
+    try {
+      const fileContents = await readFile(file, 'utf8')
+
+      // Skip empty files
+      if (!fileContents.trim()) {
+        core.debug(`Skipping empty file ${file}`)
+        continue
+      }
+
+      const response = await fetch(`${axeLinterUrl}/lint-source`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: apiKey
+        },
+        body: JSON.stringify({
+          source: fileContents,
+          filename: file,
+          config: linterConfig
+        })
+      })
+
+      const result = (await response.json()) as LinterResponse
```

Comment:
We need to handle network errors and unhealthy server responses.

```suggestion
      const response = await fetch(`${axeLinterUrl}/lint-source`, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          Authorization: apiKey
        },
        body: JSON.stringify({
          source: fileContents,
          filename: file,
          config: linterConfig
        })
      })

      if (!response.ok) {
        // ...
      }
      
      var contentType = response.headers.get("Content-Type")
      if (!contentType || !contentType.includes("json")) {
        // ...
      }

      const result = (await response.json()) as LinterResponse
```